src/dowel/csv_output.py

```diff
@@ -41,13 +41,33 @@ def record(self, data, prefix=''):
                     extrasaction='ignore')
                 self._writer.writeheader()
 
+            # new fieldnames detected, need to read in current file, modify, and write everything again
             if to_csv.keys() != self._fieldnames:
-                self._warn('Inconsistent TabularInput keys detected. '
-                           'CsvOutput keys: {}. '
-                           'TabularInput keys: {}. '
-                           'Did you change key sets after your first '
-                           'logger.log(TabularInput)?'.format(
-                               set(self._fieldnames), set(to_csv.keys())))
+                # obtain extra fieldname
+                extra_field = (set(to_csv.keys()) ^ set(self._fieldnames)).pop()
+
+                # close original log file
+                self._log_file.close()
+
+                # store original data in RAM
+                reader = csv.DictReader(open(self._log_file.name, 'r'))
+                origin_data = []
+                for row in reader:
+                    origin_data.append(row)
+
+                # reopen overwrite original log file
+                self._log_file = open(self._log_file.name, self.mode)
+                self._fieldnames = set(to_csv.keys())
+                self._writer = csv.DictWriter(
```

Comment:
It's probably best to refactor out the writer initialization logic, so that if the parameters get changed somewhere they're consistently changed.